pkg/shell/nav.jsx

```diff
@@ -0,0 +1,201 @@
+    }
+
+    onSearchChanged(e) {
+        this.setState({ search: e.target.value });
+    }
+
+    clearSearch() {
+        this.setState({ search: "" });
+    }
+
+    render() {
+        const groups = [];
+        const term = this.state.search.toLowerCase();
+        this.props.groups.forEach(g => {
+            const new_items = g.items.map(i => this.props.filtering(i, term)).filter(Boolean);
+            new_items.sort(this.props.sorting);
+            if (new_items.length > 0)
+                groups.push({ name: g.name, items: new_items });
+        });
+
+        return (
+            <>
+                <div className="has-feedback search">
+                    <input className="filter-menus form-control" type="search" placeholder={_("Search")} aria-label={_("Search")} onChange={this.onSearchChanged} value={this.state.search} />
+                    <span className="fa fa-search form-control-feedback" />
+                </div>
+                <Nav onSelect={this.onSelect} theme="dark">
+                    { groups.map(g =>
+                        <NavGroup key={g.name} title={g.name}>
+                            {g.items.map(i => this.props.item_render(i, this.state.search.toLowerCase()))}
+                        </NavGroup>
+                    )}
+                    { groups.length < 1 && <span className="non-menu-item">{_("No results found")}</span> }
+                    { this.state.search !== "" && <span className="non-menu-item"><button onClick={this.clearSearch} className="link-button hint">{_("Clear Search")}</button></span> }
+                </Nav>
+            </>
+        );
+    }
+}
+
+CockpitNav.propTypes = {
+    groups: PropTypes.array,
+    selector: PropTypes.string,
+    item_render: PropTypes.func,
+    current: PropTypes.string,
+    filtering: PropTypes.func,
+    sorting: PropTypes.func,
+};
+
+function PageStatus(props) {
+    const s = props.status;
+
+    // Generate name for the status
+    let desc = props.name.toLowerCase().split(" ");
+    desc.push(s.type);
+    desc = desc.join("-");
+
+    return (
+        <OverlayTrigger placement="right" overlay={ <Tooltip id={desc + "-tooltip"}>{ s.title }</Tooltip> }>
+            <span id={desc} className="nav-status">
+                {s.type == "error" ? <ExclamationCircleIcon color="#f54f42" />
+                    : s.type == "warning" ? <ExclamationTriangleIcon color="#f0ab00" />
+                        : <InfoCircleIcon color="#73bcf7" />}
+            </span>
+        </OverlayTrigger>
+    );
+}
+
+function FormatedText(props) {
+    function split_text(text, term) {
+        const b = text.toLowerCase().indexOf(term);
+        const e = b + term.length;
+        return [text.substring(0, b), text.substring(b, e), text.substring(e, text.length)];
+    }
+
+    const s = split_text(props.keyword, props.term);
+    return (
+        <>{s[0]}<mark>{s[1]}</mark>{s[2]}</>
+    );
+}
+
+export class CockpitNavItem extends React.Component {
```

Comment:
This is functional and thus should use the simple `const Name = ({ prop1, ... }) => JSXElement` syntax.